runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java

```diff
@@ -45,233 +48,265 @@ public LogUnitClient getClient(long epoch, UUID clusterID) {
-     * @param ctx Context
-     * @param r   Router
-     * @throws Exception Throws TrimmedException if address has already been trimmed.
+     * @param msg The write log response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router.
+     * @return Always True, since the range write was successful.
      */
-    @ClientHandler(type = CorfuMsgType.ERROR_TRIMMED)
-    private static Object handleTrimmed(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r)
-            throws Exception {
-        throw new TrimmedException();
+    @ResponseHandler(type = PayloadCase.RANGE_WRITE_LOG_RESPONSE)
+    private static Object handleRangeWriteLogResponse(ResponseMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return true;
     }
 
     /**
-     * Handle an ERROR_OVERWRITE message.
+     * Handle a read log response from the server.
      *
-     * @param msg Incoming Message
-     * @param ctx Context
-     * @param r   Router
-     * @throws OverwriteException Throws OverwriteException if address has already been written to.
+     * @param msg The read log response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router.
+     * @return {@link ReadResponse} sent back from server.
      */
-    @ClientHandler(type = CorfuMsgType.ERROR_OVERWRITE)
-    private static Object handleOverwrite(CorfuPayloadMsg<Integer> msg, ChannelHandlerContext ctx, IClientRouter r)
-            throws Exception {
-        throw new OverwriteException(OverwriteCause.fromId(msg.getPayload()));
+    @ResponseHandler(type = PayloadCase.READ_LOG_RESPONSE)
+    private static Object handleReadLogResponse(ResponseMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return getReadResponse(msg.getPayload().getReadLogResponse());
     }
 
     /**
-     * Handle an ERROR_DATA_OUTRANKED message.
+     * Handle a inspect addresses response from the server.
      *
-     * @param msg Incoming Message
-     * @param ctx Context
-     * @param r   Router
-     * @throws OverwriteException Throws OverwriteException if write has been outranked.
+     * @param msg The inspect addresses response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router.
+     * @return {@link InspectAddressesResponse} sent back from server.
      */
-    @ClientHandler(type = CorfuMsgType.ERROR_DATA_OUTRANKED)
-    private static Object handleDataOutranked(CorfuMsg msg,
-                                              ChannelHandlerContext ctx, IClientRouter r)
-            throws Exception {
-        throw new DataOutrankedException();
+    @ResponseHandler(type = PayloadCase.INSPECT_ADDRESSES_RESPONSE)
+    private static Object handleInspectResponse(ResponseMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return getInspectAddressesResponse(msg.getPayload().getInspectAddressesResponse());
```

Comment:
Most of these handlers should return the actual value, not the wrapped values. That is, LogData and not ReadResponse (which wraps LogData) etc.,